pkg/peaks/peaks.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package peaks
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"encoding/json"
+
+	"github.com/paypal/load-watcher/pkg/watcher"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	"sigs.k8s.io/scheduler-plugins/apis/config"
+	"sigs.k8s.io/scheduler-plugins/apis/config/v1beta3"
+)
+
+const (
+	Name = "Peaks"
+)
+
+type Peaks struct {
+	handle    framework.Handle
+	collector *Collector
+	args      *config.PeaksArgs
+}
+
+type PowerModel struct {
+	K0 float64 `json:"k0"`
+	K1 float64 `json:"k1"`
+	K2 float64 `json:"k2"`
+	// Power = K0 + K1 * e ^(K2 * x) : where x is utilisation
+}
+
+var _ framework.ScorePlugin = &Peaks{}
+var max_power = 0.0
+var cluster_power_model map[string]PowerModel
+
+func (pl *Peaks) Name() string {
+	return Name
+}
+
+func init_power_node_models() {
+	data, err := os.ReadFile("/power_model/node_power_model")
+	if err != nil {
+		panic(err)
+	}
+	if err = json.Unmarshal(data, &cluster_power_model); err != nil {
+        panic(err)
```

Comment:
return error instead of panicing it'll make debugging easier.

Reply:
Made the change